### PR TITLE
Find touch coords for 768x1014 Web Apps displayed on iPad Pro form factors

### PR DESCRIPTION
**Pull request by jmoody**
### Motivation

When Cordova iPad apps are displayed on iPad Pro form factors, the touch coordinates are incorrect because the layout is happening in a 768x1024 rect rather than the native resolution.

Users can avoid this by updating their apps.

* http://www.cordovacircus.com/articles/ipad-pro-and-cordova

Completes:

* tap coordinates are incorrect for some iPad form factors [VSTS](https://msmobilecenter.visualstudio.com/Mobile-Center/_workitems/edit/21628)

Includes:

* Support legacy apps displayed on iPhone X in letter box #402

### Tests

See #402

### CODE REVIEW

**Comment by jmoody** [comment]
@john7doe This change set includes the commits from #402 + 1 commit.

File: calabash/Classes/Utils/LPDevice.m [hunk 1]
```diff
@@ -291,6 +292,8 @@ - (NSDictionary *) screenDimensions {
                         @"width" : @(size.width),
                         @"scale" : @(scale),
                         @"bounds" : boundsDict,
+                        @"bounds_portrait_height" : @(screenBoundsHeight),
```
Comment (john7doe) [inline calabash/Classes/Utils/LPDevice.m]: does it make sense to also log the nativeBounds?

File: calabash/Classes/Utils/LPDevice.m [hunk 2]
```diff
@@ -512,6 +515,15 @@ - (BOOL) isLetterBox {
   }
 }
 
+- (BOOL) isIPhone10LetterBox {
+  if (![self isIPhone10Like]) { return NO; }
+  NSDictionary *dimensions = [self screenDimensions];
+  NSInteger screenModeHeight = [dimensions[@"height"] integerValue];
+  NSInteger screenBoundsHeight = [dimensions[@"bounds_portrait_height"] integerValue];
+
+  return screenModeHeight == 2001 && screenBoundsHeight == 667;
```
Comment (john7doe) [inline calabash/Classes/Utils/LPDevice.m]: Any case where `screenModeHeight > screenBoundsHeight` is per definition a letterbox case, right?

Comment (jmoody) [reply calabash/Classes/Utils/LPDevice.m]: Maybe?

I think it is possible that some apps will be scaled and not be in letter box.

File: calabash/Classes/Utils/LPDevice.m (hunk 1 above)
Comment (jmoody) [reply calabash/Classes/Utils/LPDevice.m]: What are the native bounds?

Comment (john7doe) [reply calabash/Classes/Utils/LPDevice.m]: https://developer.apple.com/documentation/uikit/uiscreen/1617810-nativebounds

Comment (jmoody) [reply calabash/Classes/Utils/LPDevice.m]: Added.

**Pull request merged**